### PR TITLE
xc7a100t support.

### PR DESCRIPTION
**Pull request by tcal-x**
Completed 'make db-part-only-artix7_100t' (fuzzers 000,001,005,074) successfully.

Completed 'make db-extras-artix7-harness' successfully.   

Builds database/artix7/xc7a100tfgg676-1/ and xc7a100tcsg324-1/.

### CODE REVIEW

**Comment by litghost** [comment]
Needs DCO on commits.

**Comment by tcal-x** [comment]
One note, this may be the first time we have multiple extra targets in a family:
`ARTIX_PARTS=artix7_200t artix7_100t`
...so I hope this doesn't have a chance of breaking things for 200t...

**Comment by litghost** [comment]
> One note, this may be the first time we have multiple extra targets in a family:
> `ARTIX_PARTS=artix7_200t artix7_100t`
> ...so I hope this doesn't have a chance of breaking things for 200t...

You can look at diff in the logs, and see that 200T didn't change.

File: fuzzers/074-dump_all/ignored_wires/artix7/xc7a100tfgg676-1_ignored_wires.txt [hunk 1]
```diff
@@ -0,0 +1,132 @@
+LIOI3_TBYTETERM_X0Y113/LIOI_I2GCLK_BOT1
```
Comment (acomodi) [inline fuzzers/074-dump_all/ignored_wires/artix7/xc7a100tfgg676-1_ignored_wires.txt:1]: I think that the ignored wires should be sorted

Comment (tcal-x) [reply fuzzers/074-dump_all/ignored_wires/artix7/xc7a100tfgg676-1_ignored_wires.txt:1]: I agree!   I made a lot of use of 'sort', 'sed', and 'grep' comparing the ignored wires against those for the 50T.

Comment (tcal-x) [reply fuzzers/074-dump_all/ignored_wires/artix7/xc7a100tfgg676-1_ignored_wires.txt]: I have sorted the wires, just using the basic alphabetical sort.  This matches what was done for the 200T part.

**Comment by acomodi** [comment]
> ...so I hope this doesn't have a chance of breaking things for 200t...

Looks like no changes happened to 200T so everything worked as supposed.

```
----------------------------------------
 Database Diff Summary
----------------------------------------
 Info.md                                       |    100 +-
 artix7/harness/arty-a7/pmod/design.bit        |    Bin 2192111 -> 2192111 bytes
 artix7/harness/arty-a7/pmod/design.dcp        |    Bin 129516 -> 129529 bytes
 artix7/harness/arty-a7/swbut/design.bit       |    Bin 2192111 -> 2192111 bytes
 artix7/harness/arty-a7/swbut/design.dcp       |    Bin 129414 -> 129404 bytes
 artix7/harness/arty-a7/uart/design.bit        |    Bin 2192111 -> 2192111 bytes
 artix7/harness/arty-a7/uart/design.dcp        |    Bin 122800 -> 122824 bytes
 artix7/harness/basys3/swbut/design.bit        |    Bin 2192111 -> 2192111 bytes
 artix7/harness/basys3/swbut/design.dcp        |    Bin 138961 -> 139255 bytes
 artix7/mask_lioi3.db                          |      2 +
 artix7/mask_lioi3_tbytesrc.db                 |      2 +
 artix7/mask_lioi3_tbyteterm.db                |      2 +
 artix7/mask_rioi3.db                          |      2 +
 artix7/mask_rioi3_tbytesrc.db                 |      2 +
 artix7/mask_rioi3_tbyteterm.db                |      2 +
 artix7/segbits_int_l.origin_info.db           |     16 +-
 artix7/segbits_int_r.origin_info.db           |     10 +-
 artix7/segbits_liob33.db                      |      1 +
 artix7/segbits_liob33.origin_info.db          |      1 +
 artix7/segbits_lioi3.db                       |      1 +
 artix7/segbits_lioi3.origin_info.db           |      1 +
 artix7/segbits_lioi3_tbytesrc.db              |      1 +
 artix7/segbits_lioi3_tbytesrc.origin_info.db  |      1 +
 artix7/segbits_lioi3_tbyteterm.db             |      1 +
 artix7/segbits_lioi3_tbyteterm.origin_info.db |      1 +
 artix7/segbits_riob33.db                      |      1 +
 artix7/segbits_riob33.origin_info.db          |      1 +
 artix7/segbits_rioi3.db                       |      1 +
 artix7/segbits_rioi3.origin_info.db           |      1 +
 artix7/segbits_rioi3_tbytesrc.db              |      1 +
 artix7/segbits_rioi3_tbytesrc.origin_info.db  |      1 +
 artix7/segbits_rioi3_tbyteterm.db             |      1 +
 artix7/segbits_rioi3_tbyteterm.origin_info.db |      1 +
 artix7/timings/DSP_L.sdf                      |     82 +-
 artix7/timings/DSP_R.sdf                      |     82 +-
 artix7/xc7a100tcsg324-1/package_pins.csv      |    213 +
 artix7/xc7a100tcsg324-1/part.json             |    771 +
 artix7/xc7a100tcsg324-1/part.yaml             |    499 +
 artix7/xc7a100tcsg324-1/tileconn.json         | 474618 ++++++++++++++++++++++
 artix7/xc7a100tcsg324-1/tilegrid.json         | 479881 +++++++++++++++++++++++
 artix7/xc7a100tfgg676-1/package_pins.csv      |    343 +
 artix7/xc7a100tfgg676-1/part.json             |    772 +
 artix7/xc7a100tfgg676-1/part.yaml             |    499 +
 artix7/xc7a100tfgg676-1/tileconn.json         | 474618 ++++++++++++++++++++++
 artix7/xc7a100tfgg676-1/tilegrid.json         | 479881 +++++++++++++++++++++++
 45 files changed, 1912256 insertions(+), 157 deletions(-)

```

**Comment by YCin** [comment]
My problem is to add a family Spartan 7, i want to add the device XC7S25-CSGA324
Thanks

**Comment by YCin** [comment]
> Completed 'make db-part-only-artix7_100t' (fuzzers 000,001,005,074) successfully.
> 
> Completed 'make db-extras-artix7-harness' successfully.
> 
> Builds database/artix7/xc7a100tfgg676-1/ and xc7a100tcsg324-1/.

how did you make that?
what tiles are you changed?
what tilegrid must be included?
thaks.

**Comment by tcal-x** [comment]
@YCin , it seems we have support for xc7s50.   First of all, you should confirm that xc7s25 is in fact a different chip.   For example, xc7a35T and xc7a50T are the same silicon, just with different software restrictions.

If they are in fact different, then try following this guide to see how to add the support for xc7s25: https://github.com/SymbiFlow/prjxray/blob/master/docs/db_dev_process/newpart.rst

**Comment by huangwenlin222** [comment]
> If they are in fact different, then try following this guide to see how to add the support for xc7s25: https://github.com/SymbiFlow/prjxray/blob/master/docs/db_dev_process/newpart.rst

Open Vivado to create the project. When selecting the chip, for Spartan7, only the xc7s50 series chips are optional, the xc7s25 series chips information cannot be found. It means that the Vivado 2017.2 software used by Prjxray only supports the xc7s50 series for Spartan7, and does not support other series such as xc7s25? 
Therefore, it is impossible to add the xc7s25 series?
Or we can use newer Vivado, instead of only Vivado2017.2 as in [Quickstart](https://f4pga.readthedocs.io/projects/prjxray/en/latest/db_dev_process/readme.html#quickstart-guide)?
Thanks a million.
图1

**Pull request merged**